crates/biome_js_analyze/src/lint/nursery/no_unknown_property.rs

```diff
@@ -0,0 +1,1266 @@
+    "className",
+    "dangerouslySetInnerHTML",
+    "defaultValue",
+    "defaultChecked",
+    "htmlFor",
+    // Events' capture events
+    "onBeforeInput",
+    "onChange",
+    "onInvalid",
+    "onReset",
+    "onTouchCancel",
+    "onTouchEnd",
+    "onTouchMove",
+    "onTouchStart",
+    "suppressContentEditableWarning",
+    "suppressHydrationWarning",
+    "onAbort",
+    "onCanPlay",
+    "onCanPlayThrough",
+    "onDurationChange",
+    "onEmptied",
+    "onEncrypted",
+    "onEnded",
+    "onLoadedData",
+    "onLoadedMetadata",
+    "onLoadStart",
+    "onPause",
+    "onPlay",
+    "onPlaying",
+    "onProgress",
+    "onRateChange",
+    "onResize",
+    "onSeeked",
+    "onSeeking",
+    "onStalled",
+    "onSuspend",
+    "onTimeUpdate",
+    "onVolumeChange",
+    "onWaiting",
+    "onCopyCapture",
+    "onCutCapture",
+    "onPasteCapture",
+    "onCompositionEndCapture",
+    "onCompositionStartCapture",
+    "onCompositionUpdateCapture",
+    "onFocusCapture",
+    "onBlurCapture",
+    "onChangeCapture",
+    "onBeforeInputCapture",
+    "onInputCapture",
+    "onResetCapture",
+    "onSubmitCapture",
+    "onInvalidCapture",
+    "onLoadCapture",
+    "onErrorCapture",
+    "onKeyDownCapture",
+    "onKeyPressCapture",
+    "onKeyUpCapture",
+    "onAbortCapture",
+    "onCanPlayCapture",
+    "onCanPlayThroughCapture",
+    "onDurationChangeCapture",
+    "onEmptiedCapture",
+    "onEncryptedCapture",
+    "onEndedCapture",
+    "onLoadedDataCapture",
+    "onLoadedMetadataCapture",
+    "onLoadStartCapture",
+    "onPauseCapture",
+    "onPlayCapture",
+    "onPlayingCapture",
+    "onProgressCapture",
+    "onRateChangeCapture",
+    "onSeekedCapture",
+    "onSeekingCapture",
+    "onStalledCapture",
+    "onSuspendCapture",
+    "onTimeUpdateCapture",
+    "onVolumeChangeCapture",
+    "onWaitingCapture",
+    "onSelectCapture",
+    "onTouchCancelCapture",
+    "onTouchEndCapture",
+    "onTouchMoveCapture",
+    "onTouchStartCapture",
+    "onScrollCapture",
+    "onWheelCapture",
+    "onAnimationEndCapture",
+    "onAnimationIteration",
+    "onAnimationStartCapture",
+    "onTransitionEndCapture",
+    "onAuxClick",
+    "onAuxClickCapture",
+    "onClickCapture",
+    "onContextMenuCapture",
+    "onDoubleClickCapture",
+    "onDragCapture",
+    "onDragEndCapture",
+    "onDragEnterCapture",
+    "onDragExitCapture",
+    "onDragLeaveCapture",
+    "onDragOverCapture",
+    "onDragStartCapture",
+    "onDropCapture",
+    "onMouseDown",
+    "onMouseDownCapture",
+    "onMouseMoveCapture",
+    "onMouseOutCapture",
+    "onMouseOverCapture",
+    "onMouseUpCapture",
+    // Video specific
+    "autoPictureInPicture",
+    "controlsList",
+    "disablePictureInPicture",
+    "disableRemotePlayback",
+];
+
+fn normalize_attribute_case(name: &str) -> &str {
+    DOM_PROPERTIES_IGNORE_CASE
+        .iter()
+        .find(|element| element.to_lowercase() == name.to_lowercase())
```

Comment:
This allocates memory. You can do a case insensitive comparison instead with [eq_ignore_ascii_case](https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case).